discord/ui/view.py

```diff
@@ -86,12 +125,20 @@ def __init__(self, children: list[Item]):
 
     def find_open_space(self, item: Item) -> int:
         for index, weight in enumerate(self.weights):
-            if weight + item.width <= 5:
+            # check if open space AND (next row has no items OR this is the last row)
+            if (weight + item.width <= 5) and (
+                (index < len(self.weights) - 1 and self.weights[index + 1] == 0)
+                or index == len(self.weights) - 1
+            ):
                 return index
 
         raise ValueError("could not find open space for item")
 
     def add_item(self, item: Item) -> None:
+        if (
+            item._underlying.is_v2() or not self.fits_legacy(item)
+        ) and not self.requires_v2():
+            self.weights.extend([0, 0, 0, 0, 0] * 7)
```

Comment:
There is no check in _ViewWeights to determine if the weights were already extended.
```py
v = discord.ui.View()
v.add_item(discord.ui.Button(label="0"))
v.add_item(discord.ui.TextDisplay("1"))
v.add_item(discord.ui.TextDisplay("2"))
print(len(v._View__weights.weights))
```
`75`
This should be `40` but was expanded by `35` again when the second cv2 component is added.